### PR TITLE
added drawing markers loop

### PR DESCRIPTION
**Pull request by zThundy**
This is a little modification of the current marker drawing system:

If the local markers table get filled with too many markers, the for iter (wich is set to cycle every frame) get filled with too many elements and has some troubles drawing markers without keeping stable performances.

This works but needs some intensive testing (wich i can't do right now). I also made a markers drawing using the gridsystem file developed by deterministic_bubble on the FiveM forum. Let me know if you want a peak on the code.

### CODE REVIEW

File: vrp/client/map.lua [hunk 1]
```diff
@@ -121,14 +122,34 @@ function tvRP.removeNamedMarker(name)
...
+-- markers sort loop
+Citizen.CreateThread(function()
+  while true do
+    Citizen.Wait(1000)
+
+    local px,py,pz = tvRP.getPosition()
+
+    for k,v in pairs(markers) do
+      -- 150 is the min distance wich the markers will be starting to be drawn
+      if GetDistanceBetweenCoords(v.x,v.y,v.z,px,py,pz,true) <= 150.0 then
```
Comment (JamesEU) [inline vrp/client/map.lua:134]: Use vidst 

#(vec3(v.x,v.y,v.z) - vec3(px,py,pz)) <= 150.0

**Review by JamesEU — CHANGES_REQUESTED** [review]
Hi, to keep up with optimizations we should use maths rather than calculating via Vdists it is much quicker just to do the math.

File: vrp/client/map.lua [hunk 2]
```diff
@@ -121,14 +122,34 @@ function tvRP.removeNamedMarker(name)
...
     Citizen.Wait(0)
 
     local px,py,pz = tvRP.getPosition()
 
-    for k,v in pairs(markers) do
+    -- if this loop get filled with too many markers, the clientside
+    -- starts lagging
+    for k,v in pairs(drawing_markers) do
       -- check visibility
       if GetDistanceBetweenCoords(v.x,v.y,v.z,px,py,pz,true) <= v.visible_distance then
```
Comment (JamesEU) [inline vrp/client/map.lua:154]: Although you didn't write this it would be nice to do the maths on this also.

**Review by JamesEU — CHANGES_REQUESTED** [review]

File: vrp/client/map.lua (hunk 2 above)
Comment (JamesEU) [reply vrp/client/map.lua:154]: Rather than `GetDistanceBetweenCoords`

Comment (zThundy) [reply vrp/client/map.lua:154]: nice idea. it was just a quick fix for markers draw. I can edit this and commit on my fork if you want

**Comment by JamesEU** [comment]
Thanks for the changes.

**Pull request merged**